### DISCUSSION TITLE
Adding explanation to checkbox lists and similar

### DISCUSSION THREAD

**Discussion by enkelmedia** [umbraco/Umbraco-CMS#11038] [Features and ideas]
When using the current checkbox-list we can enter any number of prevalues which is just a simple value.

![image](https://user-images.githubusercontent.com/1782524/132304674-38db6036-4720-473c-a1a5-098dca41eac8.png)

Rendered like this

![image](https://user-images.githubusercontent.com/1782524/132304815-5537e2fa-a096-42dc-ac50-520dbf08b287.png)

I'm proposing that we should add a description to the pre values and render this with each option as sometimes the pre value does not say enough for the editor to know what to choose.

So that we would render each checkbox with the pre value, and then a small-tag under with the description.

This would apply to at least the "checkbox list" but maybe also dropdowns (could use title-attribute for each "option" in the select maybe).

What do you think about this? I could work on a PR if this is something we want.

OR we should evaluate to add something like the Data List from Contentment https://github.com/leekelleher/umbraco-contentment to the core.

**Comment by bjarnef** [comment]
I agree a description like in Contentment can be super useful to content editors. Also related to this disqussion, where I would like we had a bit more control of the actual stored value.
https://github.com/umbraco/Umbraco-CMS/discussions/11084

I think we need a prevalue editor to handle an array of objects (label, value, description) and not just a string of string values.
Ideally configurate the properties to specific (data)types e.g. textstring, textarea, numeric, etc.
